### PR TITLE
lambda upload failures support for happy path

### PR DESCRIPTION
**Pull request by srm78**
Implement lambda upload failures support. Verified lambda is invoked and s3 objects getting cleaned up.

### CODE REVIEW

File: backend/corpora/dataset_processing/process.py [hunk 1]
```diff
@@ -226,8 +226,6 @@ def main():
         os.environ["DATASET_ID"],
         os.environ["DROPBOX_URL"],
         "local.h5ad",
-        os.environ["ARTIFACT_BUCKET"],
-        os.environ["CELLXGENE_BUCKET"],
```
Comment (srm78) [inline backend/corpora/dataset_processing/process.py:230]: These lines are added by #799. the called function `download_from_dropbox_url` is only taking 3 arguments resulting in a failure and also not using these parameters. for now I have removed these.

**Review by jgadling — APPROVED** [review]

File: backend/corpora/common/utils/aws.py [hunk 2]
```diff
@@ -98,6 +98,6 @@ def delete_many_from_s3(bucket_name: str, dataset_uuid: str) -> None:
     """
     if not dataset_uuid:
         raise ValueError
-    s3 = boto3.resource("s3")
+    s3 = boto3.resource("s3", endpoint_url=os.getenv("BOTO_ENDPOINT_URL"))
```
Comment (MDunitz) [inline backend/corpora/common/utils/aws.py:101]: will this cause errors if BOTO_ENDPOINT_URL isnt set?

**Review by MDunitz — APPROVED** [review]
one small question, otherwise lgtm

File: backend/corpora/common/utils/aws.py (hunk 2 above)
Comment (jgadling) [reply backend/corpora/common/utils/aws.py:101]: Nope, os.getenv() will return None if that env var isn't set.

**Pull request merged**